Mu2 final revision changes - ViSolve

visolve-openemr opened the pull request.

bradymiller commented:
Hi,

Please address the comments in the following previous review:
https://github.com/openemr/openemr/pull/315/files

Let me know when that is done. Then I will do another review.

thanks,
-brady

visolve-openemr commented:
@bradymiller , Sorry. We missed the previous commit comments. It has been completed. Here is the commit link https://github.com/visolve-openemr/Openemr_features_enhancements/commit/e67552f1f94bd6eaae86e1e21c3769d21ea37e79. Please review.

bradymiller commented on interface/main/messages/messages.php:
```diff
@@ -255,6 +255,23 @@
+			$pname = $prow['fname']." ".$prow['lname'];
+			echo xlt('Linked Image Report') . ":</b>\n";
+			echo "<a href='javascript:void(0);' "; 
+			echo "onClick=\"gotoReport(".$d->get_id().",'".$pname."',".$prow['pid'].",".$prow['pubpid'].",'".$str_dob."');\">";
```
need to html each variable with attr(addslashes()). example:
``````
attr(addslashes($d->get_id()))

bradymiller commented on interface/main/messages/messages.php:
```diff
@@ -409,6 +426,35 @@
+	parent.left_nav.setPatientEncounter(EncounterIdArray,EncounterDateArray,CalendarCategoryArray);
+	var baseurl  = '<?php  echo $GLOBALS['webroot'] . "/controller.php?document&view"; ?>';
+	var params   = "&patient_id=" + pid + "&document_id=" + doc_id + "&";
+	location.href = baseurl + params;
```
This may possibly need a little work after testing (to ensure it works in both frames mode and tabs mode). Will test it after next revision.

bradymiller replied:
TODO for bradymiller

bradymiller commented on interface/main/messages/messages.php:
```diff
@@ -409,6 +426,35 @@
+		url:'<?php echo $GLOBALS['webroot']."/library/ajax/set_session_ajax.php";?>',
+		data:{set_pid: pid},
+		async: false
+	});
```
Is the /library/ajax/set_session_ajax.php script and ajax call needed? There are several places in OpenEMR that change to a new patient where this is not necessary.

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -2125,6 +2125,10 @@ ALTER TABLE `users` ADD `suffix` varchar(255) default NULL;
 
+#IfMissingColumn log category
+ALTER TABLE `log` ADD `category` varchar(255) default NULL;
+#EndIf
```
Entries to this file need to go to bottom of the file.

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -2137,3 +2141,21 @@ CREATE TABLE `product_registration` (
+#IfNotTable log_validator
+CREATE TABLE `log_validator` (
+  `log_id` bigint(20) NOT NULL,
+  `log_checksum` longtext DEFAULT NULL,
```
remove default NULL here(this is not legal in newer mysql versions):
``````
`log_checksum` longtext,

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -2137,3 +2141,21 @@ CREATE TABLE `product_registration` (
+
+-- Add Image Result note type for electronic indication - a12
+#IfNotRow2D list_options list_id note_type option_id Image Results
+	SET @seq = (SELECT MAX(seq) FROM  list_options where list_id = 'note_type');
```
No need to do this and rec just making the seq 30.

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -2137,3 +2141,21 @@ CREATE TABLE `product_registration` (
+#EndIf
+
+#IfMissingColumn pnotes document_id
+	ALTER TABLE `pnotes` ADD `document_id` int(11) DEFAULT NULL;
```
It is done via the setGpRelation() function. Do a grep for it to see how it's used.

bradymiller commented on sql/database.sql:
```diff
@@ -5523,6 +5525,7 @@ CREATE TABLE `pnotes` (
   `message_status` VARCHAR(20) NOT NULL DEFAULT 'New',
   `portal_relation` VARCHAR(100) NULL,
   `is_msg_encrypted` TINYINT(2) DEFAULT '0' COMMENT 'Whether messsage encrypted 0-Not encrypted, 1-Encrypted',
+  `document_id` int(11) DEFAULT NULL,
```
As discussed, there is already a way to map document id's to pnotes. Would be best to use that rather than creating another redundant method. It is done via the setGpRelation() function. Do a grep for it to see how it's used.

bradymiller commented on sql/database.sql:
```diff
@@ -9582,3 +9587,10 @@ CREATE TABLE `product_registration` (
+DROP TABLE IF EXISTS `log_validator`;
+CREATE TABLE `log_validator` (
+  `log_id` bigint(20) NOT NULL,
+  `log_checksum` longtext DEFAULT NULL,
```
remove default NULL here(this is not legal in newer mysql versions):
``````
`log_checksum` longtext,

bradymiller commented:
Hi @visolve-openemr, 
Review is complete. Looking forward to the next revision (I plan to test that revision and as long as all comments are addressed, then should be ready to go into the codebase).
thanks, -brady

visolve-openemr commented:
@bradymiller , we have addressed all the comments and committed the code. We came to know about the gprelations table while we working on a12(Image Results), but we need to demonstrate as accessing the image result document from messages screen. So we prefer to go with new field in the pnotes table. We changed that now to use gprelations table as you suggested. Here is the new commit link https://github.com/visolve-openemr/Openemr_features_enhancements/commit/e796a249c4639b71e876eb84d521c045a7003c85. Kindly review.

visolve-openemr commented:
Additional commit link https://github.com/visolve-openemr/Openemr_features_enhancements/commit/4f57c5e710edef448710919aba11c99b1ab27f3b. Removed set_session_ajax.php script file

bradymiller commented on library/log_validation.php:
```diff
@@ -0,0 +1,62 @@
+		if(!$valid) break;
+	}
+	if($valid){
+		echo "Audit Log Validated Successfully";
```
TODO for bradymiller

bradymiller commented on sql/database.sql:
```diff
@@ -9582,3 +9586,10 @@ CREATE TABLE `product_registration` (
+DROP TABLE IF EXISTS `log_validator`;
+CREATE TABLE `log_validator` (
+  `log_id` bigint(20) NOT NULL,
+  `log_checksum` longtext NULL,
```
TODO for bradymiller

bradymiller commented on interface/main/messages/messages.php:
```diff
@@ -409,6 +410,35 @@
+		url:'<?php echo $GLOBALS['webroot']."/interface/patient_file/encounter/patient_encounter.php";?>',
+		data:{set_pid: pid},
+		async: false
+	});
```
TODO for bradymiller
I do not think this ajax call is required. Will remove and test it.

bradymiller commented:
Just brought this code into the codebase. Thanks for the MU2 contribution! -brady
